### DISCUSSION TITLE
cpu at only 50%

### DISCUSSION THREAD

**Discussion by markifi** [openai/whisper#1551] [Q&A] [answered]
i know everyone here uses gpu-s but i only have an intel 4400, besides i find the performance of the tiny model adequate enough on the cpu, however it only uses 2 cpu cores out of 4. is there a simple setting somewhere i could set to use all the cores? i'm running version 20230314. thanks for the amazing software; it's truly unbelievable. 

bonus question: where does whisper store it's config file? i'm on linux but it's not in .config

**Comment by phineas-pta** [comment]
if u dont have powerful hardware u may want try https://github.com/ggerganov/whisper.cpp

also there's no config file whatsoever

**Reply by walter8729** [reply]
whisper audio.mp3  --model medium --device cpu --threads 4

por ejemplo, para usar solo la cpu, si tienes una gpu ya no soportada.. y limitas la cantidad de hilos.

**Comment by glangford** [comment, marked as answer]
On a typical Intel processor, you might have 2 physical CPU cores that are divided into 4 virtual cores (hyperthreading). By default I believe whisper only  counts the physical cores, so it appears like you don't have full utilization, but you may or may not get better performance using hyperthreading.

You can try setting `--threads 4` from the command line to see if you get better performance; see `whisper --help`.

**Reply by markifi** [reply]
oops, i didn't see it in whisper --help before for some reason. my mistake. performance was ~0.55x-0.88x real time before with 50% cpu utilisation. a short english test file i used for this finishes in 98 seconds using --threads 2 and 119 seconds with --threads 4, and it is a dual core, you're right. so at least with this single example it's actually slower, even though cpu utilisation did go up to 100%. thank you both for replying
